test/framework/provider/wait.go

```diff
@@ -52,27 +65,41 @@ func WaitForAllProvidersInstalled(ctx context.Context, c client.Client, interval
 }
 
 // Wait for Provider to be successfully updated.
-func WaitForRevisionTransition(ctx context.Context, c client.Client, p2 string, p1 string, interval time.Duration, timeout time.Duration) error {
+func WaitForRevisionTransition(ctx context.Context, c client.Client, p2 string, p1 string, interval time.Duration, timeout time.Duration, t *testing.T) error {
 	if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
 		l := &v1.ProviderRevisionList{}
 		if err := c.List(ctx, l); err != nil {
 			return false, err
 		}
+		// t.Logf("Provider Revisions list: %+v", l.Items)
+		for _, item := range l.Items {
+			t.Logf("Name: %v \t Type: %v", item.Name, item.TypeMeta)
+		}
```

Comment:
@parul5sahoo given that we are looping through items in cases below where we return `false` (i.e. we will keep polling), we could avoid doing so here as the later logs will inform us of the status that is causing us to continue polling.

Reply:
Got it! Done!

Reply:
Should I do the same for [these lines](https://github.com/parul5sahoo/test/blob/6eb3d299a5de602da3e9f1b8dd452bbb25395419/test/framework/provider/wait.go#L37-L39) as well?